app/server/lib/DocWorkerLoadTracker.ts

```diff
@@ -99,16 +114,92 @@ export class DocWorkerLoadTracker {
+   * This file should contain a value in bytes, the function will convert it
+   * to mega bytes.
+   *
+   * @param filePath The path to the file to read
+   * @param valueProcessor A function that may return an amount of memory if the file contains a special value.
+   *
+   * @returns The amount of memory reported by the file converted to megabytes.
+   */
+  private async _readValueFromFileInMB(
+    filePath: string,
+    valueProcessor?: (val: string) => number|undefined
+  ): Promise<number> {
+    const rawVal = await fs.readFile(filePath, "utf-8");
+    const valInBytes = valueProcessor?.(rawVal) ?? parseInt(rawVal, 10);
+
+    if (isNaN(valInBytes)) {
+      throw new Error(
+        `Unexpected value (not a number) found in file in "${filePath}". value = ${rawVal.slice(0, 1000)}`
+      );
+    }
+
+    return valInBytes / 1024**2;
+  }
+
+  private _canReadValueFromFile(filePath: string|undefined): filePath is string {
+    return filePath !== undefined && !this._disabledFilePaths.has(filePath);
+  }
+
+  /**
+   * We read the memory used in this order:
+   * 1. If we have a path specified for a file that contain the memory used, read this file
+   * 2. Otherwise read instead the load using the estimation given by the doc manager
+   *    (less accurate, typically it does not include nodejs load).
+   *
+   * @returns The used memory
+   * @throws When the file at the path can't be read or doesn't contain a number.
+   */
+  private async _getMemoryUsedMB(): Promise<number> {
+    if (this._canReadValueFromFile(Deps.docWorkerUsedMemoryBytesPath)) {
+      return await this._readValueFromFileInMB(Deps.docWorkerUsedMemoryBytesPath);
+    }
+
+    return  this._docManager.getTotalMemoryUsedMB();
+  }
+
+  /**
+   * We read the total memory available in this order:
+   * 1. If the admin specified an amount of total memory available through GRIST_DOC_WORKER_MAX_MEMORY_MB
+   *    then use it (to cover the case the administrator wants to pass a lower value than the actual
+   *    total memory, and have spare free memory for the current documents)
+   * 2. If the admin specified a path to read the total amoun of memory, read it.
+   * 2.1. If the value is max, consider as "Infinity"
+   * 2.2. If the value is a number, return it
+   * 3. Return Infinity
+   *
+   * @returns The total memory we should consider as available for the worker.
+   * @throws When the file at the path can't be read or doesn't contain a valid value as described above.
+   */
+  private async _getMemoryTotalMB() {
+    if (Deps.docWorkerMaxMemoryMBForcedValue !== undefined) {
+      return Deps.docWorkerMaxMemoryMBForcedValue;
+    }
+    if (this._canReadValueFromFile(Deps.docWorkerMaxMemoryBytesPath)) {
+      return await this._readValueFromFileInMB(
+        Deps.docWorkerMaxMemoryBytesPath,
+        // When the value is "max", return Infinity, otherwise return undefined so
+        // so the function read what's probably an integer value.
```

Comment:
```suggestion
        // When the value is "max", return Infinity, otherwise return undefined
        // so the function reads what's probably an integer value.
```